mzLib/Proteomics/RetentionTimePrediction/Chronologer/ChronologerRetentionTimeEstimator.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TorchSharp;
+
+namespace Proteomics.RetentionTimePrediction.Chronologer;
+public static class ChronologerRetentionTimeEstimator
+{
+    /// <summary>
+    /// Uses the Chronologer model to predict the retention time of a peptide (reported in % of ACN).
+    /// Only modifications present in the Chronologer dictionary are supported.
+    /// <code>
+    /// "Carbamidomethyl on C"
+    /// "Oxidation on M"
+    /// "Glu to PyroGlu"
+    /// "Phosphorylation on S"
+    /// "Phosphorylation on T"
+    /// "Phosphorylation on Y"
+    /// "Accetylation on K"
+    /// "Succinylation on K"
+    /// "Ubiquitination on K"
+    /// "Methylation on K"
+    /// "Dimethylation on K"
+    /// "Trimethylation on K"
+    /// "Methylation on R"
+    /// "Dimethylation on R"
+    /// </code>
+    /// </summary>
+    /// <param name="baseSequence"></param>
+    /// <param name="fullPeptideSequence"></param>
+    /// <returns></returns>
+    public static double PredictRetentionTime(string baseSequence, string fullPeptideSequence)
+    {
+        torch.Tensor tensor = Tensorize(baseSequence, fullPeptideSequence);
+        var model = new Chronologer();
```

Comment:
Does a new model need to be constructed every time a prediction is made? This feels very resource intensive to load those weights every time. Predictions are called for every PSM and Peptide in MM. Would it make more sense to have a static instance constructed and just call that?

Reply:
The SSRCalc file is the RetentionTimePrediction folder.

Yes, I am currently working on a static class, so no re-loading weights is required.